ocean_lib/ocean/ocean_pool.py

```diff
@@ -655,22 +659,27 @@ def _get_all_liquidity_records(
         token_address=None,
         raw_result=True,
     ):
-        action_to_fn = {
-            "join": "get_join_logs",
-            "exit": "get_exit_logs",
-            "swap": "get_swap_logs",
-        }
         current_block = to_block if to_block is not None else self.web3.eth.block_number
         pool = BPool(self.web3, pool_address)
         dt_address = token_address or self.get_token_address(pool_address, pool)
-        factory = DTFactory(self.web3, get_dtfactory_address(web3=self.web3))
+        factory = DTFactory(self.web3, self.dtfactory_address)
```

Comment:
Summary: I'm ok with this change. 

Details: When I first saw the chnage, I thought: why not just keep `get_dtfactory_address()` approach and add in the missing argument `address_file`? Since it avoids redundant storage of information. But then digging around, it's clear that it's gonna add complexities to get the `address_file` value. And there are questions of "which chain" too. By the time you've solved it, you've got something much more complex.

Whereas if you simply store the `dtfactory_address` attribute then everything stays clean. And it follows a pattern identical to the `bfactory_address` attribute.

It's the latter that's implemented here. Bravo!

Reply:
My thoughts exactly! :)